backend/src/test/java/com/woowacourse/kkogkkog/domain/CouponStatusTest.java

```diff
@@ -5,141 +5,163 @@
 
 import com.woowacourse.kkogkkog.exception.InvalidRequestException;
 import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
+@DisplayName("CouponStatus 클래스의")
 class CouponStatusTest {
 
-    @Test
-    @DisplayName("READY 상태의 쿠폰은 REQUEST 이벤트를 받으면 REQUESTED 상태로 변경된다.")
-    void readyChangesToRequestedOnRequestEvent() {
-        CouponStatus currentStatus = CouponStatus.READY;
-        CouponEvent event = CouponEvent.REQUEST;
-
-        CouponStatus actual = currentStatus.handle(event);
-        CouponStatus expected = CouponStatus.REQUESTED;
-
-        assertThat(actual).isEqualTo(expected);
-    }
-
-    @Test
-    @DisplayName("REQUESTED 상태의 쿠폰은 REQUEST 이벤트를 받으면 예외가 발생된다.")
-    void requestedCanNotHandleRequestEvent() {
-        CouponStatus currentStatus = CouponStatus.REQUESTED;
-        CouponEvent event = CouponEvent.REQUEST;
-
-        assertThatThrownBy(() -> currentStatus.handle(event))
-            .isInstanceOf(InvalidRequestException.class);
+    @Nested
+    @DisplayName("READY 상태는")
+    class Ready {
```

Comment:
공유해주신 자료 읽어봤는데 일단 nested여도 테스트 클래스명은 `~Test` 형식으로 통일하는게 맞지 않나 싶어요. [junit 자료](https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested)의 경우 모든 클래스들이 `~Test` 컨벤션을 맞추지 않기는 하는데, 애초에 main 패키지 없이 test 패키지만 있다는 가정 하에서 작성된 문서가 아닌가 싶어요. 

특히 해당 클래스들(`Ready`, `Requested` 등)이 자동 Import 대상으로 나타날 때 끝에 Test가 안붙어있으면 뇌정지가 올 수 있지 않나 싶어요.

<img width="559" alt="image" src="https://user-images.githubusercontent.com/73531614/181154016-6be2dea6-3e28-40cc-8db0-000d5978ebe5.png">

```suggestion
    @Nested
    @DisplayName("READY 상태는")
    class ReadyTest {
```

Reply:
https://www.testwithspring.com/lesson/the-best-practices-of-nested-unit-tests/

Reply:
https://www.testwithspring.com/lesson/the-best-practices-of-nested-unit-tests/

Reply:
https://canro91.github.io/2021/04/12/UnitTestNamingConventions/
https://www.testwithspring.com/lesson/the-best-practices-of-nested-unit-tests/

NestedTest 자체가 레퍼런스가 별로 없긴한데, 구글에서 검색해서 나온 결과물들은 대부분 Test를 붙이지 않는것을 제시하고 있긴합니다. 딱 정해져있는 사항이라 생각되지는 않아서 팀끼리 Test를 붙이는게 더 낫다고 생각한다면 팀컨벤션으로 맞추는 것도 괜찮다고 생각하기는 합니다. 다만 저는 Test를 붙이는 일이 귀찮고 효용이 없어보인다는게 제일 거슬리긴합니다.

추가로 클래스는 대문자로 시작하고, 메서드는 소문자로 시작하는 컨벤션을 지키고 있으니 제시해주신 사진처럼 메서드를 사용하다 뇌정지 올일이 별로 없을거라는 생각입니다.

추가로 상관없는 내용이긴 하지만 스프링 문서에 있는 Test관련 내용에도 `@Nested` 에는 딱히 안붙히고 있기는 합니다.

<img width="660" alt="Screen Shot 2022-07-27 at 12 55 25 PM" src="https://user-images.githubusercontent.com/75936123/181157451-676dc05f-7be6-4e7b-948d-329954bb81ed.png">